doc/userguide/setting-up-ipsinline-for-linux.rst

```diff
@@ -115,12 +118,134 @@ There is also a way to use iptables with multiple networks (and interface cards)
+
+ suricata -q 3 -q 4 -q 5
+
+`fanout` and `bypass` are the two available options:
+
+- `fanout`: When used together with load balancing, this will use the CPU ID instead of connection hash as an index to map packets to the queues. The idea is that you can improve performance if there’s a queue per CPU. This requires total with a number of queues superior to 1 to be specified.
+- `bypass`: By default, if no userspace program is listening on an Netfilter queue, then all packets that are to be queued are dropped. When this option is used, the queue rule behaves like ACCEPT if there is no program listening, and the packet will move on to the next table.
+
+The `bypass` option can be used to avoid downtime of link when Suricata is not running but this also means that the blocking feature will not be present.
+
+Settings up IPS at Layer 2
+--------------------------
+
+AF_PACKET IPS mode
+~~~~~~~~~~~~~~~~~~
+
+AF_PACKET capture method is supporting a IPS/Tap mode. In this mode, you just need the interfaces to be up. Suricata will take care of copying the packets from one interface to the other. No iptables or nftables configuration is necessary.
+
+You need to dedicate two network interfaces for this mode. The configuration is made via configuration variable available in the description of an AF_PACKET interface.
+
+For example, the following configuration will create a Suricata acting as IPS between interface eth0 and eth1: ::
+
+ af-packet:
+   - interface: eth0
+     threads: 1
+     defrag: no
+     cluster-type: cluster_flow
+     cluster-id: 98
+     copy-mode: ips
+     copy-iface: eth1
+     buffer-size: 64535
+     use-mmap: yes
+   - interface: eth1
+     threads: 1
+     cluster-id: 97
+     defrag: no
+     cluster-type: cluster_flow
+     copy-mode: ips
+     copy-iface: eth0
+     buffer-size: 64535
+     use-mmap: yes
+
+Basically, we’ve got an af-packet configuration with two interfaces. Interface eth0 will copy all received packets to eth1 because of the copy-* configuration variable ::
+
+    copy-mode: ips
+    copy-iface: eth1
+
+The configuration on eth1 is symmetric ::
+
+    copy-mode: ips
+    copy-iface: eth0
+
+There is some important points to consider when setting up this mode:
+
+- The implementation of this mode is dependent of the zero copy mode of AF_PACKET. Thus you need to set `use-mmap` to yes on both interface.
+- MTU on both interfaces have to be equal: the copy from one interface to the other is direct and packet bigger then the MTU will be dropped by kernel.
+- Set different values of cluster-id on both interfaces to avoid conflict.
+- Any network card offloading creating bigger then physical layer datagram will result in packets drop as transmit path can not handle them.
+- Set `stream.inline` to `yes` so Suricata really switches to blocking mode
```

Comment:
Ppl can leave it at 'auto', which is the default.

Reply:
outch, forgot about this code in AFP...